Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```
@@ -0,0 +1,212 @@
+
+    if (!$ProjectGuidAndFilePathMap.ContainsKey($guidToken))
+    {
+        Write-Verbose "GetPathRelativeToRepoRoot: An entry for project guid $guidToken could not be found, check ProjectInfo.xml file"
+        return $null
+    }
+    if (!$ComponentKeyAndPathMap.ContainsKey($component))
+    {
+        Write-Verbose "GetPathRelativeToRepoRoot: An entry for component key $component could not be found, check sonar-report.json file"
+        return $null
+    }
+
+    #This stores the full on-disk path of the *.xxproj file
+    $projectPath = $($ProjectGuidAndFilePathMap[$guidToken])
+
+    $finalFilePath = [System.IO.Path]::GetDirectoryName($projectPath)
+    
+    $finalFilePath = [System.IO.Path]::Combine($finalFilePath, $ComponentKeyAndPathMap[$component])
+    Write-Verbose "GetPathRelativeToRepoRoot: finalFilePath:$finalFilePath"
+
+    $repoLocalPath = GetTaskContextVariable "Build.Repository.LocalPath"
+    
+    if (!$repoLocalPath)
+    {
+        throw "GetPathRelativeToRepoRoot: Could not get task variable Build.Repository.LocalPath"
+    }
+
+    Write-Verbose "GetPathRelativeToRepoRoot: repoLocalPath:$repoLocalPath"
+
+    #this will remove from the file path, the part upto the repo name. 
+    #e.g. finalFilePath=C:\Agent\_work\ef030e14\s\Mail2Bug\Main.cs and repoLocalPath=C:\Agent\_work\ef030e14\s
+    #after the SubString() call finalFilePath=\Mail2Bug\Main.cs
+    $finalFilePath = $finalFilePath.ToString().SubString($repoLocalPath.Length);
+
+    #Replace '\' with '/'. VSO expects file path like /Mail2Bug/Main.cs (\Mail2Bug\Main.cs does not work)
+    $finalFilePath = $finalFilePath.ToString().Replace([System.IO.Path]::DirectorySeparatorChar, [System.IO.Path]::AltDirectorySeparatorChar)
+    Write-Verbose "GetPathRelativeToRepoRoot: Returning finalFilePath:$finalFilePath"
+
+    #save data into cache so next time we don't have to compute
+    $ComponentKeyAndRelativePathCache.Add($component, $finalFilePath)
```

Ok, so it seems that it can't be null. 
Might as well throw at the top instead of check if it's null or not in the first condition.